Initial application setup

rmroot opened the pull request:
connects #1 

Initial application setup:
- Angular v16.2.0
- Electron v27.0.2
- Bootstrap v5.3.2

Added static navbar and welcome page.
Added initial API for communication between Angular and Electron through Electron Service
Updated README with instructions for running application

jordanperr commented on angular.json:
```diff
@@ -0,0 +1,104 @@
+    }
+  },
+  "cli": {
+    "analytics": "75e308bc-dccd-4ffa-a72d-8f37f42a6f09"
```
What is the purpose of this hash? Looks like it is used to configure Angular's CLI analytics. Is it standard practice to distribute this key with Angualr applications?

Also, the Angular CLI is not currently a dependency of this project.

rmroot replied:
Set to false to turn off cli analtics.

jordanperr commented on README.md:
```diff
@@ -0,0 +1,46 @@
+
+- `npm run dist` will create electron installers for your operating system
+
+- Installer will be created in an `/output/neb-tool/` directory in the **parent** directory you run the command in
```
Why is the parent directory specified?

rmroot replied:
Update package.json build instructions to output to ./output/neb-tool

jordanperr commented on README.md:
```diff
@@ -0,0 +1,46 @@
+
+## Running unit tests
+
+Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.io).
```
Angular CLI is not specified as a dependency, so we should add some lines about how to install it. There also may be situations where users do not have root access to their machine, and we could point to documentation describing how to install it into ~/.local

jordanperr commented:
Thank you, mark, for uploading the first draft. This is a great starting point for us! I was able to run everything but the tests, which require the `ng` command line tool. I'll revisit this at a later time.

My comments, in addition to the inline comments.
- [ ] I made a slight changes the readme (as discussed) creating a dependency section and noting our NPM version is 20.9 LTS and updating a path.
- [ ] Warning when building (npm run build and build-prod) "bundle initial exceeded maximum budget. Budget 500.00 kB was not met by 12.93 kB with a total of 512.93 kB." Do we care about this?
- [ ] Various deprecation warnings when running npm install (see below) Do we care about this?

Deprecation warnings:
- npm WARN deprecated @babel/plugin-proposal-unicode-property-regex@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-unicode-property-regex instead.
- npm WARN deprecated @babel/plugin-proposal-async-generator-functions@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-async-generator-functions instead.
- npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
- npm WARN deprecated @wessberg/ts-evaluator@0.0.27: this package has been renamed to ts-evaluator. Please install ts-evaluator instead

rmroot commented:
> Thank you, mark, for uploading the first draft. This is a great starting point for us! I was able to run everything but the tests, which require the `ng` command line tool. I'll revisit this at a later time.
> 
> My comments, in addition to the inline comments.
> 
> * [ ]  I made a slight changes the readme (as discussed) creating a dependency section and noting our NPM version is 20.9 LTS and updating a path.
> * [ ]  Warning when building (npm run build and build-prod) "bundle initial exceeded maximum budget. Budget 500.00 kB was not met by 12.93 kB with a total of 512.93 kB." Do we care about this?
> * [ ]  Various deprecation warnings when running npm install (see below) Do we care about this?
> 
> Deprecation warnings:
> 
> * npm WARN deprecated @babel/plugin-proposal-unicode-property-regex@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-unicode-property-regex instead.
> * npm WARN deprecated @babel/plugin-proposal-async-generator-functions@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-async-generator-functions instead.
> * npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
> * npm WARN deprecated @wessberg/ts-evaluator@0.0.27: this package has been renamed to ts-evaluator. Please install ts-evaluator instead

I have addressed all of the comments with the exception of adding how to install the angular CLI to the README. Still need to get to that. 

I did remove codesigning related things for the mac installer in package.json at this time because we don't have that in place and it's a whole beast in itself.

Don't believe we care about the "Budget" we can just bump it up. And I am not concerned with the deprecation warnings at this time.

jordanperr commented on README.md:
```diff
@@ -0,0 +1,50 @@
+- `npm run dist` will create electron installers for your operating system
+
+- Installer will be created in an `./output/neb-tool/` directory 
+
```
Right now the tests do not run without Google chrome installed. It looks like there are configuration files for Karma that allow users to specify which browsers to test on: https://karma-runner.github.io/6.4/config/configuration-file.html

I think we should either include these files and instructions on how to set up the test environment, or add some note about it here in the readme.

rmroot replied:
Updated README with note on chrome dependency and link to Karma config docs.

jordanperr reviewed (approved):
I'm able to reproduce the example application by following the instructions in the Readme, including dependency installation, running tests through Karma. I think this will be a great base from which we can develop NEB Tool. Thanks for iterating with me on some of the details in the Readme.
